tests/PHPStan/Rules/Operators/data/bcmath-number.php

```diff
@@ -0,0 +1,406 @@
+		assertType('*ERROR*', $a ^ $b);
+		assertType('*ERROR*', $a | $b);
+		assertType('bool', $a && $b);
+		assertType('true', $a || $b);
+		assertType('bool', $a and $b);
+		assertType('bool', $a xor $b);
+		assertType('true', $a or $b);
+	}
+
+	public function bcVsCallable(Number $a, callable $b): void
+	{
+		assertType('*ERROR*', $a + $b);
+		assertType('*ERROR*', $a - $b);
+		assertType('*ERROR*', $a * $b);
+		assertType('*ERROR*', $a / $b);
+		assertType('*ERROR*', $a % $b);
+		assertType('*ERROR*', $a . $b);
+		assertType('*ERROR*', $a ** $b);
+		assertType('*ERROR*', $a << $b);
+		assertType('*ERROR*', $a >> $b);
+		assertType('bool', $a < $b);
+		assertType('bool', $a <= $b);
+		assertType('bool', $a > $b);
+		assertType('bool', $a >= $b);
+		assertType('int<-1, 1>', $a <=> $b);
+		assertType('bool', $a == $b);
+		assertType('bool', $a != $b);
+		assertType('*ERROR*', $a & $b);
+		assertType('*ERROR*', $a ^ $b);
+		assertType('*ERROR*', $a | $b);
+		assertType('bool', $a && $b);
+		assertType('true', $a || $b);
+		assertType('bool', $a and $b);
+		assertType('bool', $a xor $b);
+		assertType('true', $a or $b);
+	}
+
+	public function bcVsIterable(Number $a, iterable $b): void
+	{
+		assertType('*ERROR*', $a + $b);
+		assertType('*ERROR*', $a - $b);
+		assertType('*ERROR*', $a * $b);
+		assertType('*ERROR*', $a / $b);
+		assertType('*ERROR*', $a % $b);
+		assertType('*ERROR*', $a . $b);
+		assertType('*ERROR*', $a ** $b);
+		assertType('*ERROR*', $a << $b);
+		assertType('*ERROR*', $a >> $b);
+		assertType('bool', $a < $b);
+		assertType('bool', $a <= $b);
+		assertType('bool', $a > $b);
+		assertType('bool', $a >= $b);
+		assertType('int<-1, 1>', $a <=> $b);
+		assertType('bool', $a == $b);
+		assertType('bool', $a != $b);
+		assertType('*ERROR*', $a & $b);
+		assertType('*ERROR*', $a ^ $b);
+		assertType('*ERROR*', $a | $b);
+		assertType('bool', $a && $b);
+		assertType('bool', $a || $b);
+		assertType('bool', $a and $b);
+		assertType('bool', $a xor $b);
+		assertType('bool', $a or $b);
+	}
+
+	public function bcVsStringable(Number $a, \Stringable $b): void
+	{
+		assertType('*ERROR*', $a + $b);
+		assertType('*ERROR*', $a - $b);
+		assertType('*ERROR*', $a * $b);
+		assertType('*ERROR*', $a / $b);
+		assertType('*ERROR*', $a % $b);
+		assertType('non-empty-string', $a . $b);
+		assertType('*ERROR*', $a ** $b);
+		assertType('*ERROR*', $a << $b);
+		assertType('*ERROR*', $a >> $b);
+		assertType('bool', $a < $b);
+		assertType('bool', $a <= $b);
+		assertType('bool', $a > $b);
+		assertType('bool', $a >= $b);
+		assertType('int<-1, 1>', $a <=> $b);
+		assertType('bool', $a == $b);
+		assertType('bool', $a != $b);
+		assertType('*ERROR*', $a & $b);
+		assertType('*ERROR*', $a ^ $b);
+		assertType('*ERROR*', $a | $b);
+		assertType('bool', $a && $b);
+		assertType('true', $a || $b);
+		assertType('bool', $a and $b);
+		assertType('bool', $a xor $b);
+		assertType('true', $a or $b);
+	}
+
+	public function bcVsNever(Number $a): void
```

Comment:
There's a bunch of inconsistencies in how never type is handled. I assume that all of these operations should result in `*NEVER*`, but I'm ignoring it for this PR.